app/src/main/java/com/realityexpander/tasky/presentation/login_screen/LoginScreen.kt

```diff
@@ -1,22 +1,174 @@
 package com.realityexpander.tasky.presentation.login_screen
 
+import androidx.activity.compose.BackHandler
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.Button
+import androidx.compose.material.CircularProgressIndicator
 import androidx.compose.material.Text
-import androidx.compose.material.TextField
-import androidx.compose.runtime.Composable
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Alignment.Companion.CenterVertically
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.unit.dp
 import androidx.hilt.navigation.compose.hiltViewModel
 import com.ramcosta.composedestinations.annotation.Destination
+import com.ramcosta.composedestinations.navigation.DestinationsNavigator
+import com.realityexpander.tasky.presentation.components.EmailField
+import com.realityexpander.tasky.presentation.components.PasswordField
+import com.realityexpander.tasky.presentation.destinations.RegisterScreenDestination
+import kotlinx.coroutines.launch
 
 @Composable
 @Destination(start = true)
 fun LoginScreen(
-    viewModel: LoginViewModel = hiltViewModel()
+    email: String? = null,
+    password: String? = null,
+    confirmPassword: String? = null,
+    navigator: DestinationsNavigator,
+    viewModel: LoginViewModel = hiltViewModel(),
 ) {
+    BackHandler(true) { /* We want to disable back clicks */ }
 
-    Text(text = "Login")
+    val loginState by viewModel.loginState.collectAsState()
+    val scope = rememberCoroutineScope()
+    val focusManager = LocalFocusManager.current
+
+    fun performLogin() {
+        scope.launch {
```

Comment:
Is there a reason you chose the UI scope over `viewModelScope`? 
What if there is a config change during execution of this function?

Reply:
Good catch, I was wondering about this. I will move any viewModel specific coroutines to the ViewModel's `.sendEvent` function. I will keep Compose-specific scopes in the Composable.

💯